### PR TITLE
size optimising bits and pieces

### PR DESCRIPTION
**Pull request by peterferrie**
two possible bugs, search for";;pf:"

### CODE REVIEW

File: Platform/Apple/virtual/src/font/fontEngine.s [hunk 1]
```diff
@@ -1497,30 +1483,22 @@ Flg_Prm2 !byte $00	;Parameter digit 1st/2nd
...
-	JMP Wp_Tab
-Wpr_06	CMP #7 		;7=Change Char/Ticker Rate
+GetParm	TXA		;restore alpha char
+	DEY
+	CPY #2 		;1=Foreground color / 2=Background color
+	BCS Wpr_02	;no - keep looking
+	STY Flg_FBc	;clear Fore/Bkgnd clr flag
+	BCC Wp_StClr	;set color
+;and carry is set from here down
+Wpr_02	DEY 		;3=Special Character
```
Comment (martinhaye) [inline Platform/Apple/virtual/src/font/fontEngine.s]: Had to add an extra `DEY` here.

File: Platform/Apple/virtual/src/font/fontEngine.s [hunk 2]
```diff
@@ -1573,9 +1544,7 @@ Wp_eChr1 CLC
 	JMP PlotFnt	;plot the character
 
 ;Change Font
-Wp_CFnt	TXA 		;restore alpha char
-	SEC
-	SBC #$30	;change Chr"#" to Val#
+Wp_CFnt	SBC #$30	;change Chr"#" to Val#
 	AND #$03	;mask off digit
 ;;pf: this CMP is broken
```
Comment (martinhaye) [inline Platform/Apple/virtual/src/font/fontEngine.s]: True. In fact, it appears we don't support change-font at all (nor need we). Removed.

**Comment by peterferrie** [comment]
> Something in the following part of the patch is causing very strange font
> rendering. Working on identifying the exact problem.

Since I've broken it, I should fix it.
Since you've started already, let me know if you want me to check it, too.

**Comment by martinhaye** [comment]
I found the problem Peter, it was a missing `DEY`. I'll take you up on your offer in the future, if you go on future byte-slaying expeditions.

Thanks again!

**Pull request merged**